homeassistant/components/recorder/services.py

```diff
@@ -35,15 +36,31 @@
 ATTR_DOMAINS = "domains"
 ATTR_ENTITY_GLOBS = "entity_globs"
 
-SERVICE_PURGE_ENTITIES_SCHEMA = vol.Schema(
-    {
-        vol.Optional(ATTR_DOMAINS, default=[]): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(ATTR_ENTITY_GLOBS, default=[]): vol.All(
-            cv.ensure_list, [cv.string]
-        ),
-        vol.Optional(ATTR_KEEP_DAYS, default=0): cv.positive_int,
-    }
-).extend(cv.ENTITY_SERVICE_FIELDS)
+
+def __invalidKey(value: Any) -> None:
+    raise vol.Invalid("key not allowed")
+
+
+SERVICE_PURGE_ENTITIES_SCHEMA = (
+    vol.Schema(
+        {
+            vol.Optional(ATTR_DOMAINS, default=[]): vol.All(
+                cv.ensure_list, [cv.string]
+            ),
+            vol.Optional(ATTR_ENTITY_GLOBS, default=[]): vol.All(
+                cv.ensure_list, [cv.string]
+            ),
+            vol.Optional(ATTR_KEEP_DAYS, default=0): cv.positive_int,
+        }
+    )
+    .extend(cv.ENTITY_SERVICE_FIELDS)
```

Comment:
This ain't an entity service, so that is where it goes wrong.

Reply:
I would have switched the target selector for an entity selector, but then I don't know how to meet this requirement:

> Note: The entity_id is only optional when used in automations.yaml or scripts.yaml . When using the UI to call this service then it is mandatory to specify at least one entity_id using the Target Picker or via YAML mode.

An entity selector as a field won't require a user to pick an entity, unless it is marked required, in which case I think it's no longer optional in automations.